### ISSUE TITLE
Feature: implement SeriesGroupBy.{value_counts, unique}

### ISSUE THREAD

**Issue by h-vetinari** [dask/dask#5146]
I'm trying to break some custom aggregation functions down into chunks / agg / finalize as in the [docs](http://docs.dask.org/en/latest/dataframe-groupby.html?highlight=set_index%20disk#aggregate), but I'm stumbling over the fact that `SeriesGroupBy` in dask (as of v2.1.0) cannot perform `value_counts()`.

This seems like an operation that could be easily distributed (each partition does the value counts separately, and then all that's necessary is adding the chunk-results together).

Admittedly, this can cause a lot of traffic if the Series contains only unique values (i.e. every value_count is 1), but that's just the nature of the beast.

Similarly, `SeriesGroupby.unique` could be implemented. This is orders of magnitude slower in pandas than the (pandas-)`value_counts`, but is conceptually even simpler then `value_counts` - rather than count the appeareances per value, we just need to keep track which values appeared at all.

Since `pd.SeriesGroupBy.unique` performs so much worse than `pd.SeriesGroupBy.value_counts`, it would even suffice to implement `dask.SeriesGroupBy.unique` through `dask.SeriesGroupBy.value_counts` (essentially by discarding the count).

**Comment by jcrist** [comment]
I agree that this would be a useful addition. Would you be interested in submitting a PR?

**Comment by h-vetinari** [comment]
@jcrist 
I'll try to have a look. Could you point me in the direction where this should be implemented, and maybe after what method I could model it?

**Comment by jcrist** [comment]
Sure. The relevant code is this class: https://github.com/dask/dask/blob/2c7ac8808ead66d4c217948f9693d7b4039e66ae/dask/dataframe/groupby.py#L1557

You'll be adding a new reduction method. Reductions are usually created using the `_aca_agg` private method - "aca" stands for "apply-concatenate-apply", which describes the parallelism model (one function is applied to all the chunks, the results are concatenated, then another function is applied to the results). The trick here will be writing these two methods. See the other reductions for some examples.

**Comment by stsievert** [comment]
How is this different than the existing implementation? In

**Comment by arpit1997** [comment]
@jcrist Is this issue resolved? Is there any issue with the PR above?

**Comment by stsievert** [comment]
@arpit1997 I'm the author of that PR, and jotted down some of the issues in see https://github.com/dask/dask/pull/5358#issuecomment-531515639
